ex1/buttonActions.js

```diff
@@ -1,13 +1,50 @@
 
-var indexOfColor = 1;
-function changeColor() {
-  var colorArray = ['#FF6633', '#CC9999','#4D80CC'];
-    //store
-    sessionStorage.setItem("color", colorArray[indexOfColor]);
-    // Retrieve
-    document.getElementById("demo").style.backgroundColor = sessionStorage.getItem("color");
-
-    if(++indexOfColor == colorArray.length){
-      indexOfColor = 0;
-    }
+const INITIAL_COLOR = 0;
+const colorArray = ['blue', 'red','orange','yellow'];
+let indexOfColor = INITIAL_COLOR;
+let weGotToEndOfTheArrayFlag = false;
+
+function handleButtonAndImageStyles(){
+   handleButtonColor();
+   handleImageDisplay();
+}
+
+function handleButtonColor(){
+  setButtonColor();
+  if(++indexOfColor == colorArray.length){
+    weGotToEndOfTheArrayFlag = true;
+    indexOfColor = INITIAL_COLOR;
+  }
+}
+
+function handleImageDisplay(){
+  let noaImageElement = document.getElementById("noa-img");
+  if(weGotToEndOfTheArrayFlag){
+    noaImageElement.style.display = "block";
+    weGotToEndOfTheArrayFlag = false;
+  } else {
+    noaImageElement.style.display = "none";
+  }
+}
+
+
+//A function to start the array colors from the start
+function restartArray() {
+  indexOfColor = INITIAL_COLOR;
+  handleButtonAndImageStyles();
+}
+
+//A function which sets the DOM
+document.addEventListener("DOMContentLoaded", function(event) {
+  handleButtonAndImageStyles();
+});
+
+
+function setButtonColor(){
+  // setButtonColor
+  let currentButtonColor = colorArray[indexOfColor];
+  //store
+  window.localStorage.setItem("color", currentButtonColor);
+  // Retrieve
+  document.getElementById("buttonID").style.backgroundColor = window.localStorage.getItem("color");;
```

Comment:
I understand now why this doesn't work for you.  only problem is you kept saying that localStorage doesn't work on your computer and I advise you to start being humble from now and never say that a basic function in a library/language doesn't work on your computer, rather than saying: I don't know how to use it or I couldn't figure out how it works.

Reply:
look at the flow of this function:
you first check what is the index and get the current color.
then you set in on the localStorage.
then you take the color from the localStorage.

So every time the program starts, you take the INITIAL_COLOR and set the indexOfColor and then set it on the localStorage and take that color from the localStorage.
Instead of doing this, set the indexOfColor on page load only if you have no localStorage.
There's more but I'll let you try to figure it out from here.
I advise you to use breakpoints and debug the flow, it'll help you understand exactly where these kind of bugs are.  check in Google how to debug on chrome

Reply:
> I understand now why this doesn't work for you. only problem is you kept saying that localStorage doesn't work on your computer and I advise you to start being humble from now and never say that a basic function in a library/language doesn't work on your computer, rather than saying: I don't know how to use it or I couldn't figure out how it works.

i didn't say that it doesnt work on my computer...I said that?